
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>1. Flower Surveys &#8212; The wildflowers in Biel</title>
    
  <link href="_static/css/theme.css" rel="stylesheet" />
  <link href="_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/sphinx-book-theme.5f77b4aec8189eecf79907ce328c390d.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebe@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="2. Wildflower germination" href="germination.html" />
    <link rel="prev" title="Biel‚Äôs flowers" href="intro.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
      
      <h1 class="site-logo" id="site-title">The wildflowers in Biel</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   Biel‚Äôs flowers
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  The Flower Survey
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   1. Flower Surveys
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Germination Experiment
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="germination.html">
   2. Wildflower germination
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Conclusion
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="resources.html">
   3. Conclusion
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Further reading
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="references.html">
   4. Bibliografie
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/analysis.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/thorerismann/flora-buckets"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/thorerismann/flora-buckets/issues/new?title=Issue%20on%20page%20%2Fanalysis.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/thorerismann/flora-buckets/master?urlpath=tree/docs/analysis.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#objectives">
   1.1. Objectives üçÄ
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#background">
   1.2. Background ‚ùÅ
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#crisis-of-biodiversity-in-vascular-plants">
     1.2.1. Crisis of Biodiversity in vascular plants ‚öò
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#middle-plateau">
     1.2.2. Middle plateau üèû
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#urban-and-suburban-conservation">
     1.2.3. Urban and suburban conservation üåπ
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#survey-locations">
   1.3. Survey locations
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#method">
   1.4. Method  üçÉ
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#results">
   1.5. Results üìà
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#categorizing-the-species">
     1.5.1. Categorizing the species üåº
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#non-invasive-neophytes">
     1.5.2. Non-invasive neophytes üå∑
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#invasive-species">
     1.5.3. Invasive species  üå∫
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#redlist-and-priority-list-species-on-the-middle-plateau">
     1.5.4. Redlist and priority list species on the middle plateau  üå∏
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#discussion-and-conclusion">
   1.6. Discussion and conclusion üíê
  </a>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="flower-surveys">
<h1><span class="section-number">1. </span>Flower Surveys<a class="headerlink" href="#flower-surveys" title="Permalink to this headline">¬∂</a></h1>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Final notebook for 2020-2021 plant surveys</span>

<span class="c1"># math and data packages</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">math</span>
<span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">ndimage</span>

<span class="c1"># charting and graphics</span>
<span class="kn">import</span> <span class="nn">matplotlib</span> <span class="k">as</span> <span class="nn">mpl</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="c1"># from matplotlib import colors</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
<span class="kn">import</span> <span class="nn">matplotlib.ticker</span> <span class="k">as</span> <span class="nn">ticker</span>
<span class="kn">import</span> <span class="nn">matplotlib.dates</span> <span class="k">as</span> <span class="nn">mdates</span>
<span class="kn">from</span> <span class="nn">matplotlib.gridspec</span> <span class="kn">import</span> <span class="n">GridSpec</span>
<span class="kn">import</span> <span class="nn">matplotlib.image</span> <span class="k">as</span> <span class="nn">mpimg</span>

<span class="c1"># os and file types</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">datetime</span> <span class="k">as</span> <span class="nn">dt</span>
<span class="kn">import</span> <span class="nn">json</span>
<span class="kn">import</span> <span class="nn">csv</span>

<span class="c1"># images and display</span>
<span class="kn">import</span> <span class="nn">base64</span><span class="o">,</span> <span class="nn">io</span><span class="o">,</span> <span class="nn">IPython</span>
<span class="kn">from</span> <span class="nn">PIL</span> <span class="kn">import</span> <span class="n">Image</span> <span class="k">as</span> <span class="n">PILImage</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">Markdown</span> <span class="k">as</span> <span class="n">md</span>
<span class="kn">from</span> <span class="nn">IPython.display</span> <span class="kn">import</span> <span class="n">display</span>

<span class="kn">import</span> <span class="nn">botany_utilities</span> <span class="k">as</span> <span class="nn">bot</span>

<span class="c1"># import survey data and initialize datetime</span>
<span class="n">spd</span><span class="o">=</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">&quot;resources/survey_data_lists_02_22.csv&quot;</span><span class="p">)</span>
<span class="n">spd</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">to_datetime</span><span class="p">(</span><span class="n">spd</span><span class="p">[</span><span class="s1">&#39;date&#39;</span><span class="p">],</span> <span class="n">errors</span><span class="o">=</span><span class="s1">&#39;coerce&#39;</span><span class="p">,</span> <span class="nb">format</span><span class="o">=</span><span class="s1">&#39;%Y-%m-</span><span class="si">%d</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">spd</span><span class="p">[</span><span class="s2">&quot;month&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">spd</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">dt</span><span class="o">.</span><span class="n">month</span>

<span class="n">charac</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;inv&quot;</span><span class="p">,</span> <span class="s2">&quot;ns&quot;</span><span class="p">,</span> <span class="s2">&quot;rch&quot;</span><span class="p">,</span> <span class="s2">&quot;ni&quot;</span><span class="p">,</span> <span class="s2">&quot;ongen&quot;</span><span class="p">]</span>

<span class="n">spd_allstats</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">bot</span><span class="o">.</span><span class="n">mutually_exclusive_plant_charac</span><span class="p">(</span><span class="n">spd</span><span class="p">,</span> <span class="n">old_col</span><span class="o">=</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">charac</span><span class="p">])</span>

<span class="n">spd1</span> <span class="o">=</span> <span class="n">spd_allstats</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">spd_allstats</span><span class="o">.</span><span class="n">year</span> <span class="o">==</span> <span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">spd2</span> <span class="o">=</span> <span class="n">spd_allstats</span><span class="o">.</span><span class="n">loc</span><span class="p">[(</span><span class="n">spd_allstats</span><span class="p">[</span><span class="s2">&quot;year&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="mi">2</span><span class="p">)]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">animage</span> <span class="o">=</span> <span class="n">bot</span><span class="o">.</span><span class="n">display_image_ipython</span><span class="p">(</span><span class="s2">&quot;resources/images/20210403_123904.jpg&quot;</span><span class="p">,</span> <span class="n">thumb</span><span class="o">=</span><span class="p">(</span><span class="mi">1200</span><span class="p">,</span><span class="mi">800</span><span class="p">),</span> <span class="n">rotates</span><span class="o">=-</span><span class="mi">90</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">animage</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_1_0.png" src="_images/analysis_1_0.png" />
</div>
</div>
<p><em><strong>Figure 1 :</strong> Chelidonum majus, a common roadside plant in Biel/Bienne, growing in the wall surrounding Nidau castle</em></p>
<div class="section" id="objectives">
<h2><span class="section-number">1.1. </span>Objectives üçÄ<a class="headerlink" href="#objectives" title="Permalink to this headline">¬∂</a></h2>
<p>The objective of this report was to construct an initial dataset on the flowering plant populations growing in different green spaces around the city of Biel/Bienne. The first survey period between June and September 2020 focused on the Suze river and its Madretschkanal variant in Biel/Bienne. The second set of surveys were conducted between February 2021 and September 2021 and included a wider geographical scope around the municipality of Biel/Bienne, namely parks, railway and road berms, empty lots and small urban forests.</p>
</div>
<div class="section" id="background">
<h2><span class="section-number">1.2. </span>Background ‚ùÅ<a class="headerlink" href="#background" title="Permalink to this headline">¬∂</a></h2>
<div class="section" id="crisis-of-biodiversity-in-vascular-plants">
<h3><span class="section-number">1.2.1. </span>Crisis of Biodiversity in vascular plants ‚öò<a class="headerlink" href="#crisis-of-biodiversity-in-vascular-plants" title="Permalink to this headline">¬∂</a></h3>
<p>There is a worldwide biodiversity crisis that will only accelerate under the increasing impacts of climate change and anthropogenic pressures. Switzerland is not an exception to these unfortunate developments, as experts have noted with increasing alarm for some time. In 2010, the IUCN Redlist report on the conservation status of Swiss vascular plants concluded that:</p>
<blockquote>
<div><p>Overall, every single Redlist documents a manifest and sustained loss of species diversity at national level. Most threatened species appear to be bound for further losses ‚Äì with no likelihood of this trend being reversed in the short term. In other words, the requirement enshrined in the Swiss Constitution that animal and plant species be prevented from going extinct has not been fulfilled.</p>
</div></blockquote>
<p>Ten years later, after identifying widespread local extinctions that irreversibly harm genetic diversity of plant populations and precede general extinctions, Swiss researchers concluded <span id="id1">[<a class="reference internal" href="references.html#id9">DM <em>et al.</em>, 2020</a>]</span>:</p>
<blockquote>
<div><p>Our study presents clear evidence that current efforts to conserve threatened plant species are insufficient to achieve national and international targets (Convention on Biological Diversity (CBD), 2011; Swiss Biodiversity Strategy 2012) for maintaining biodiversity. The current paradigm of protecting and restoring threatened habitats is failing to avert extinctions. Going forward, we need to develop a comprehensive landscape approach, involving the creation of ecological infrastructure and translocation and assisted migration of threatened species into suitable habitats.</p>
</div></blockquote>
<p>In Switzerland the vascular plants that are most threatened are aquatic or semi-aquatic <span id="id2">[<a class="reference internal" href="references.html#id5">Bornand <em>et al.</em>, 2016</a>]</span>. Widespread modification of waterways, the spread of invasive species, agricultural run-off, development of shorelines and regulation of water levels has led to the decline of suitable habitats and subsequently some 70% of species are evaluated as threatened to some extent in the 2016 Redlist.</p>
<p>Next, are species associated with traditional forms of agriculture and land management, namely species in low altitude prairies, ruderal areas, and cropland/vineyards, which have largely disappeared in the past 100 years, which range from 40% - 60% threatened (Ibid). The decline of these species follows directly from changing agricultural practices. The species of least concern are associated with alpine habitat, rockfalls, and other largely undisturbed habitat that have historically been relatively spared from encroachment (Ibid). All of this is unfolding against the backdrop of accelerating climate change that researchers expect will provoke a major shift in habitat distributions and be a medium-term stress factor for a significant number of Switzerland‚Äôs vascular plants <span id="id3">[<a class="reference internal" href="references.html#id21">Vittoz <em>et al.</em>, 2013</a>]</span>.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">bot</span><span class="o">.</span><span class="n">display_image_ipython</span><span class="p">(</span><span class="s2">&quot;resources/images/redlist-species_cropped.png&quot;</span><span class="p">,</span> <span class="n">thumb</span><span class="o">=</span><span class="p">(</span><span class="mi">500</span><span class="p">,</span><span class="mi">400</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_5_0.png" src="_images/analysis_5_0.png" />
</div>
</div>
<p><em><strong>Figure 2 :</strong> This figure taken from Infoflora‚Äôs 2019 Regional Redlist shows the assessed threat level of the different plants in each of Switzerland‚Äôs regions. We see that the middle plateau / central plateau (MP) has the highest number of plants rated higher than ‚ÄúLC‚Äù, or least concern, followed by the Jura (JU).</em></p>
</div>
<div class="section" id="middle-plateau">
<h3><span class="section-number">1.2.2. </span>Middle plateau üèû<a class="headerlink" href="#middle-plateau" title="Permalink to this headline">¬∂</a></h3>
<p>The heart of Switzerland‚Äôs population and economic output, the Middle Plateau (MP) is also where localized extinctions are most prominent <span id="id4">[<a class="reference internal" href="references.html#id9">DM <em>et al.</em>, 2020</a>]</span>. Some 60% of the vascular plant species evaluated for the IUCN‚Äôs Redlist in the MP are classified as potentially or actually regionally threatened; next comes the Jura region with 50% threatened regionally <span id="id5">[<a class="reference internal" href="references.html#id8">Bornand <em>et al.</em>, 2019</a>]</span>. Again, species associated with freshwater habitats are the group of most concern, with some 3/4 of species under potential threat. Around 60% of the alpine populations that survive throughout the plateau are evaluated as under potential threat, as are just over half of lowland grassland species, and nearly half of forest species (Ibid). Increased levels of output and consumption, as well as relatively intensive agricultural practices, are identified as the main drivers of risk for regional biodiversity. Not all of this is due directly to higher anthropogenic pressures: the central plateau hosts isolated, localized populations that are inherently more vulnerable to pressures on their populations (Ibid).</p>
</div>
<div class="section" id="urban-and-suburban-conservation">
<h3><span class="section-number">1.2.3. </span>Urban and suburban conservation üåπ<a class="headerlink" href="#urban-and-suburban-conservation" title="Permalink to this headline">¬∂</a></h3>
<p>A judicious use of public spaces, such as road verges, railroad berms and parks, that balances their potential as habitats with the needs and requirements of the city‚Äôs inhabitants can mitigate some anthropogenic pressure. Research has shown that increasing the size of green patches and ensuring connectivity with larger green spaces within urban areas can boost biodiversity within cities <span id="id6">[<a class="reference internal" href="references.html#id3">Vega, 2020</a>]</span>. Allowing spaces to go unmowed, sowing native wild-flower beds, and creating micro-habitats throughout urban and suburban areas can benefit both herbaceous and predatory insects <span id="id7">[<a class="reference internal" href="references.html#id28">Sattler <em>et al.</em>, 2011</a>]</span>. In fact, (sub)urban green spaces and road verges can provide plants, insects, and birds with suitable conditions to pursue their life cycles and even become a ‚Äúlast refuge‚Äù for certain species (ibid).  While private gardens provide refuge for native species, research also shows that private gardens are heavily biased towards non-native plants, potentially making public spaces even more important as a suitable refuge for remnant wild species <span id="id8">[]</span>. As such, improvement and maintenance of areas in and around urban centers could yield outsize results in terms of biodiversity preservation at this critical juncture for the future of ecosystem services; the most recent data of nationwide biodiversity provided by the Biodiversity Monitoring System shows a decline in urban vascular plant species diversity <span id="id9">[<a class="reference internal" href="references.html#id25">for¬†the Environment, 2021</a>]</span>.</p>
<p>There are a number of different strategies that municipalities can use to boost wildflower diversity. For example, two comparable strategies are the sowing of annual wildflowers and the planting and maintenance of (native) perennials. A comparative study of these management mechanisms outcomes showed the strategy of annual patches was strongly dominated by native perennial patches, with remnant populations providing resources during the crucial early months of Spring that were not being provided by the plants seeded by the greenspace managers <span id="id10">[]</span>. Annual wildflower strips ploughed during the winter period can act as an ecological trap for overwintering arthropods who are killed during the process: one study found nearly 70% of beetles failed to emerge from hibernation in wildflower strips managed in this way<span id="id11">[<a class="reference internal" href="references.html#id29">Ganser <em>et al.</em>, 2019</a>]</span>. Perennial wildflower strips on the other hand were found to provide significant habitat for arthropods (Ibid). In short, management decisions can have wide ranging effects: one recent finding from Z√ºrich, which confirms previous research, is that sowing of commercial wildflower seed in two annual species significantly impacted the gene pool of the species in the city <span id="id12">[<a class="reference internal" href="references.html#id3">Vega, 2020</a>]</span>.</p>
</div>
</div>
<div class="section" id="survey-locations">
<h2><span class="section-number">1.3. </span>Survey locations<a class="headerlink" href="#survey-locations" title="Permalink to this headline">¬∂</a></h2>
<p>In the first year, sites were chosen almost exclusively on the urban Suze river and its Madretschkanal variant in Biel/Bienne. Along this stretch of the river there are many varied habitats, ranging from deep shade to full pavement to flower strips maintained for ecological and aesthetic purposes. Sites were chosen in a semi random way out of locations that are managed the municipality to a greater or less degree and reasonably accessible. In 2021 the surveys included a wider geographic scope, but still focused on the Madretschkanal and the urban Suze river.  Their is a bias towards locations around the Madretsch quarter and towards locations with visible flowering plants - sites with no flowers were not recorded. The additional zones surveyed ranged from railroad berms to boxes planted by the city to renaturalization sites. As such, the sites varied greatly in size, shape and species composition.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">bot</span><span class="o">.</span><span class="n">display_image_ipython</span><span class="p">(</span><span class="s2">&quot;resources/images/qgismap_one.jpeg&quot;</span><span class="p">,</span> <span class="n">thumb</span><span class="o">=</span><span class="p">(</span><span class="mi">900</span><span class="p">,</span><span class="mi">600</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_10_0.png" src="_images/analysis_10_0.png" />
</div>
</div>
<p><em><strong>Figure 3 :</strong> Map of survey locations around Biel/Bienne in 2020 - 2021. Note that some surveys occured outside the scope of this map. Note also the heavy focus on the Suze river and the Madretschkanal branch, as well as the Madretsch quarter of Biel/Bienne.</em></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">open_these_images</span><span class="p">(</span><span class="n">anarray</span><span class="p">):</span>
    <span class="k">return</span> <span class="p">[</span><span class="n">PILImage</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">.</span><span class="n">convert</span><span class="p">(</span><span class="s2">&quot;RGBA&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">anarray</span><span class="p">]</span>

<span class="k">def</span> <span class="nf">display_grid_of_images</span><span class="p">(</span><span class="n">anarray</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">rows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="p">[],</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span> <span class="p">):</span>
    <span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">rows</span><span class="p">,</span><span class="n">cols</span><span class="p">,</span><span class="n">figsize</span> <span class="o">=</span> <span class="n">fig_size</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">anax</span> <span class="ow">in</span> <span class="n">ax</span><span class="p">:</span>
        <span class="n">anax</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
    
    <span class="k">for</span> <span class="n">i</span><span class="p">,</span><span class="n">im</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">anarray</span><span class="p">):</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">imshow</span><span class="p">(</span><span class="n">im</span><span class="p">)</span>
        <span class="n">ax</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="n">labels</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">16</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;left&quot;</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">15</span><span class="p">)</span>
    
    <span class="n">plt</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
    <span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># generate images of survey sites</span>

<span class="n">images_to_display</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;resources/images/DSC04657.JPG&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/suze-2.JPG&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/DSC02009.JPG&#39;</span>
    
<span class="p">]</span>

<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s2">&quot;Ile de la Suze&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Holunderweg&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Cygnes Lot&quot;</span><span class="p">,</span>
    
<span class="p">]</span>

<span class="n">rows</span> <span class="o">=</span> <span class="mi">1</span>
<span class="n">cols</span> <span class="o">=</span> <span class="mi">3</span>
<span class="n">fig_size</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span><span class="mi">10</span><span class="p">)</span>

<span class="c1"># open the images</span>
<span class="n">anarray</span><span class="o">=</span><span class="n">open_these_images</span><span class="p">(</span><span class="n">images_to_display</span><span class="p">)</span>

<span class="c1"># display the images</span>
<span class="n">display_grid_of_images</span><span class="p">(</span><span class="n">anarray</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="n">cols</span><span class="p">,</span> <span class="n">rows</span><span class="o">=</span><span class="n">rows</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="n">labels</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="n">fig_size</span> <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_13_0.png" src="_images/analysis_13_0.png" />
</div>
</div>
<p><em><strong>Figure 4 :</strong> Images of three survey sites along the urban Suze river and the Madretschkanal. There was a great diversity in survey site characteristics amenable to many varieties of plants.</em></p>
<p>The size of the sites was not recorded, but a detailed description is available and it is also possible to recover the boundary of the sites based on the field notes and the geometry of the object as surveys were usually bounded by sidewalks, roads, or other barriers. Larger features, such as those in the photos above, were typically broken into several smaller sites and surveyed individually. Othertimes, survey sites are small and isolated, an ‚Äúisland‚Äù in a ‚Äúsea‚Äù of concrete, as in the features surveyed in Zukunftstrasse shown below on the right.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">images_to_display</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;resources/images/Selection_034.png&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/Selection_035.png&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/Selection_037.png&#39;</span>
    
<span class="p">]</span>

<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s2">&quot;Fl√∂sserweg lot&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Zukunftstrasse/Rue de l&#39;avenir&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Alleestrasse / Rue de l&#39;all√©e&quot;</span><span class="p">,</span>
    
<span class="p">]</span>

<span class="n">rows</span> <span class="o">=</span> <span class="mi">1</span>
<span class="n">cols</span> <span class="o">=</span> <span class="mi">3</span>
<span class="n">fig_size</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>

<span class="c1"># open the images</span>
<span class="n">anarray</span><span class="o">=</span><span class="n">open_these_images</span><span class="p">(</span><span class="n">images_to_display</span><span class="p">)</span>

<span class="c1"># display the images</span>
<span class="n">display_grid_of_images</span><span class="p">(</span><span class="n">anarray</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="n">cols</span><span class="p">,</span> <span class="n">rows</span><span class="o">=</span><span class="n">rows</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="n">labels</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="n">fig_size</span> <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_16_0.png" src="_images/analysis_16_0.png" />
</div>
</div>
<p><em><strong>Figure 5 :</strong> Three examples of the decomposition of city features into survey sites. <strong>Left :</strong> Shows an example of how one lot was surveyed. <strong>Middle :</strong> Two different feature types were surveyed at this location near Zukunftstrasse: Three raised garden beds put in by the city to slow street traffic, and two isolated river bank sites. <strong>Right :</strong> The curvature in the Madretschkanal and the concrete driveway crossing the canal separate adjascent survey sites.</em></p>
</div>
<div class="section" id="method">
<h2><span class="section-number">1.4. </span>Method  üçÉ<a class="headerlink" href="#method" title="Permalink to this headline">¬∂</a></h2>
<p>A site was inspected for the vascular plants in flower, and all the different species identified were recorded. Throughout the survey period, the majority of plants identified were in flower in order to increase the reliability of observations. The exceptions are a small number of tree identifications, shrub identifications (such as the genus Nitida, Taraxacum), and <em>Hedera helix</em>, which is imminently recognizable and is common throughout survey locations. Most of these non-flower identifications are to the genus level only. Note that Poaceae and Juncacea (grasses and sedges) were left off the survey entirely due to the difficulty in identification. For all species, the subspecies were not recorded. The below algorithm was repeated for all flowering plants on a survey site:</p>
<ul class="simple">
<li><p>If a plant was known it was checked for the recognizable characteristics and immediately recorded</p></li>
<li><p>If a flower was not known it was looked up in a plant identification application (Flora Incognita and/or PlantNet) and subsequently checked against the latest Flora Helvetica guidebook. If an identification was confident, it was recorded.</p></li>
<li><p>If an identification was not clear even after these consultations, it was recorded to the genus level if possible, or left off of the observations.</p></li>
</ul>
<p>Nearly all plants in flower could be identified following this algorithm, which became increasingly efficient as time went on. In June 2020, the number of plants that could be reliably identified by the surveyor was small, whereas by August 2020, the surveyor could reliably identify the common species that dominate the landscape. This meant that the 2020 - 2021 observations are not directly comparable as the surveyor was able to record a significantly higher number of species during the second year. In addition, as the number of species that were recognizable without the need to consult an algorithm or a guidebook increased, the surveys became increasingly quick. Some direct results of this data collection protocol are the following:</p>
<ul class="simple">
<li><p>The year on year results should be compared with caution as the surveyor was able to identify more plants in the second year</p></li>
<li><p>If a plant was visible but not flowering, it was very unlikely to be recorded on the survey</p></li>
<li><p>There is no information about the relative densities of the plants within surveys</p></li>
</ul>
</div>
<div class="section" id="results">
<h2><span class="section-number">1.5. </span>Results üìà<a class="headerlink" href="#results" title="Permalink to this headline">¬∂</a></h2>
<p>The following figures provide summary data from year one and year two. The general takeaway is that throughout the year there is a high level of diversity on the species, genus and family level, which reached 66 families, 275 different genus and 552 different species identified in the second year. Note that only the families of native species were included, meaning that a number of neophyte (non-native) families are not represented in the below tables.</p>
<p>While some of these surveys were conducted outside of Biel, their part of the surveys is about 7%, meaning the overall patterns remain the same with their exclusion. When compared to the observations reported to Infoflora there are 1207 different vascular plant species that have been identified in the 100 square kilometers centered on Biel/Bienne. Of these species, some 450 were identified on the plant surveys and about 100 species were identified in the surveys but not included in Infoflora‚Äôs list of observations. While a great part of this mismatch is due to the surveyor not identifying most trees, grasses and ferns, it also indicates a significant floral diversity in the region that was not identified in the surveys. It also indicates that Infoflora may underestimate the diversity of plant species in urban spaces, even if this diversity may be less valuable ecologically and come from commercial seeds and non-native plants.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># axone monthly summary values</span>
<span class="n">t_one</span> <span class="o">=</span> <span class="n">spd1</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;month&quot;</span><span class="p">],</span> <span class="n">as_index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span><span class="o">.</span><span class="n">agg</span><span class="p">({</span><span class="s2">&quot;species&quot;</span><span class="p">:</span><span class="s2">&quot;nunique&quot;</span><span class="p">,</span> <span class="s2">&quot;loc_date&quot;</span><span class="p">:</span><span class="s2">&quot;nunique&quot;</span><span class="p">,</span> <span class="s2">&quot;locname&quot;</span><span class="p">:</span><span class="s2">&quot;nunique&quot;</span><span class="p">})</span>
<span class="n">species1</span> <span class="o">=</span> <span class="n">spd1</span><span class="o">.</span><span class="n">species</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span>
<span class="n">speciesint</span> <span class="o">=</span> <span class="n">spd1</span><span class="p">[</span><span class="s2">&quot;species&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="n">species2</span> <span class="o">=</span> <span class="n">speciesint</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span>
<span class="n">species_corrected</span> <span class="o">=</span> <span class="n">species1</span><span class="o">-</span><span class="n">species2</span>

<span class="n">data</span> <span class="o">=</span> <span class="n">t_one</span><span class="o">.</span><span class="n">values</span>
<span class="n">colLabels</span> <span class="o">=</span> <span class="p">[</span><span class="o">*</span><span class="n">t_one</span><span class="o">.</span><span class="n">columns</span><span class="p">[:</span><span class="o">-</span><span class="mi">2</span><span class="p">],</span> <span class="s2">&quot;# samples&quot;</span><span class="p">,</span> <span class="s2">&quot;# places&quot;</span><span class="p">]</span>
<span class="n">colWidths</span> <span class="o">=</span> <span class="p">[</span><span class="o">*</span><span class="p">[</span><span class="mi">1</span><span class="o">/</span><span class="nb">len</span><span class="p">(</span><span class="n">colLabels</span><span class="p">)]</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">colLabels</span><span class="p">)]</span>
<span class="n">bbox</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>

<span class="c1"># axtwo species identified per sample</span>
<span class="n">d1</span> <span class="o">=</span> <span class="n">spd1</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;date&quot;</span><span class="p">,</span> <span class="s2">&quot;locname&quot;</span><span class="p">],</span> <span class="n">as_index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span><span class="o">.</span><span class="n">species</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>

<span class="c1"># axthree summary values</span>
<span class="n">summary</span><span class="o">=</span><span class="p">{</span>
    <span class="s2">&quot;number of samples&quot;</span><span class="p">:</span><span class="n">spd1</span><span class="o">.</span><span class="n">loc_date</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
    <span class="s2">&quot;first sample&quot;</span><span class="p">:</span> <span class="n">spd1</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span><span class="o">.</span><span class="n">date</span><span class="p">(),</span>
    <span class="s2">&quot;last sample&quot;</span><span class="p">:</span><span class="n">spd1</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span><span class="o">.</span><span class="n">date</span><span class="p">(),</span>
    <span class="s2">&quot;number of places&quot;</span><span class="p">:</span><span class="n">spd1</span><span class="o">.</span><span class="n">locname</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
    <span class="s2">&quot;number of species&quot;</span><span class="p">:</span><span class="n">species_corrected</span><span class="p">,</span>
    <span class="s2">&quot;number of genus&quot;</span><span class="p">:</span><span class="n">spd1</span><span class="o">.</span><span class="n">genus</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
    <span class="s2">&quot;number of families&quot;</span><span class="p">:</span><span class="n">spd1</span><span class="o">.</span><span class="n">fam</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
<span class="p">}</span>

<span class="n">a_summary</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_dict</span><span class="p">(</span><span class="n">summary</span><span class="p">,</span> <span class="n">orient</span><span class="o">=</span><span class="s2">&quot;index&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>


<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">7</span><span class="p">))</span>

<span class="n">gs</span> <span class="o">=</span> <span class="n">GridSpec</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>

<span class="n">axone</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">])</span>
<span class="n">axtwo</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">:])</span>
<span class="n">axthree</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">6</span><span class="p">:,</span> <span class="mi">6</span><span class="p">:])</span>

<span class="c1"># axone</span>
<span class="n">table_one</span> <span class="o">=</span> <span class="n">bot</span><span class="o">.</span><span class="n">a_simple_formatted_table</span><span class="p">(</span><span class="n">axone</span><span class="p">,</span><span class="n">data</span><span class="p">,</span><span class="n">colLabels</span><span class="o">=</span><span class="n">colLabels</span><span class="p">,</span> <span class="n">colWidths</span><span class="o">=</span><span class="n">colWidths</span><span class="p">,</span> <span class="n">bbox</span><span class="o">=</span><span class="n">bbox</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>

<span class="c1"># axtwo</span>
<span class="n">locator</span> <span class="o">=</span> <span class="n">mdates</span><span class="o">.</span><span class="n">MonthLocator</span><span class="p">()</span>  <span class="c1"># every month</span>
<span class="n">fmt</span> <span class="o">=</span> <span class="n">mdates</span><span class="o">.</span><span class="n">DateFormatter</span><span class="p">(</span><span class="s1">&#39;%b&#39;</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">d1</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s2">&quot;date&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;species&quot;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axtwo</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">locator</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">fmt</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;# species indentified&quot;</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="p">)</span>

<span class="c1"># axthree</span>
<span class="n">b</span> <span class="o">=</span> <span class="n">bot</span><span class="o">.</span><span class="n">a_single_column_table</span><span class="p">(</span><span class="n">axthree</span><span class="p">,</span> <span class="n">a_summary</span><span class="o">.</span><span class="n">values</span><span class="p">,</span> <span class="n">colWidths</span><span class="o">=</span><span class="p">[</span><span class="mf">.6</span><span class="p">,</span> <span class="mf">.4</span><span class="p">])</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_20_0.png" src="_images/analysis_20_0.png" />
</div>
</div>
<p><em><strong>Figure 6</strong>: Summary statistics for the first year of surveys. <strong>Left :</strong> The number of unique species, unique samples, and unique sites surveyed per month in year 1. <strong>Top right</strong>: The number of species identified per survey over year 1. <strong>Bottom right</strong>: Summary statistics of year 1.</em></p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># axone monthly summary values</span>
<span class="n">t_one</span> <span class="o">=</span> <span class="n">spd2</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;month&quot;</span><span class="p">],</span> <span class="n">as_index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span><span class="o">.</span><span class="n">agg</span><span class="p">({</span><span class="s2">&quot;species&quot;</span><span class="p">:</span><span class="s2">&quot;nunique&quot;</span><span class="p">,</span> <span class="s2">&quot;loc_date&quot;</span><span class="p">:</span><span class="s2">&quot;nunique&quot;</span><span class="p">,</span> <span class="s2">&quot;locname&quot;</span><span class="p">:</span><span class="s2">&quot;nunique&quot;</span><span class="p">})</span>
<span class="n">species1</span> <span class="o">=</span> <span class="n">spd2</span><span class="o">.</span><span class="n">species</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span>
<span class="n">speciesint</span> <span class="o">=</span> <span class="n">spd2</span><span class="p">[</span><span class="s2">&quot;species&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="n">species2</span> <span class="o">=</span> <span class="n">speciesint</span><span class="o">.</span><span class="n">nunique</span><span class="p">()</span>
<span class="n">species_corrected</span> <span class="o">=</span> <span class="n">species1</span><span class="o">-</span><span class="n">species2</span>

<span class="n">data</span> <span class="o">=</span> <span class="n">t_one</span><span class="o">.</span><span class="n">values</span>
<span class="n">colLabels</span> <span class="o">=</span> <span class="p">[</span><span class="o">*</span><span class="n">t_one</span><span class="o">.</span><span class="n">columns</span><span class="p">[:</span><span class="o">-</span><span class="mi">2</span><span class="p">],</span> <span class="s2">&quot;# samples&quot;</span><span class="p">,</span> <span class="s2">&quot;# places&quot;</span><span class="p">]</span>
<span class="n">colWidths</span> <span class="o">=</span> <span class="p">[</span><span class="o">*</span><span class="p">[</span><span class="mi">1</span><span class="o">/</span><span class="nb">len</span><span class="p">(</span><span class="n">colLabels</span><span class="p">)]</span><span class="o">*</span><span class="nb">len</span><span class="p">(</span><span class="n">colLabels</span><span class="p">)]</span>
<span class="n">bbox</span> <span class="o">=</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>

<span class="c1"># axtwo species identified per sample</span>
<span class="n">d1</span> <span class="o">=</span> <span class="n">spd2</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;date&quot;</span><span class="p">,</span> <span class="s2">&quot;locname&quot;</span><span class="p">],</span> <span class="n">as_index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span><span class="o">.</span><span class="n">species</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>

<span class="c1"># axthree summary values</span>
<span class="n">summary</span><span class="o">=</span><span class="p">{</span>
    <span class="s2">&quot;number of samples&quot;</span><span class="p">:</span><span class="n">spd2</span><span class="o">.</span><span class="n">loc_date</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
    <span class="s2">&quot;first sample&quot;</span><span class="p">:</span> <span class="n">spd2</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">min</span><span class="p">()</span><span class="o">.</span><span class="n">date</span><span class="p">(),</span>
    <span class="s2">&quot;last sample&quot;</span><span class="p">:</span><span class="n">spd2</span><span class="p">[</span><span class="s2">&quot;date&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">max</span><span class="p">()</span><span class="o">.</span><span class="n">date</span><span class="p">(),</span>
    <span class="s2">&quot;number of places&quot;</span><span class="p">:</span><span class="n">spd2</span><span class="o">.</span><span class="n">locname</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
    <span class="s2">&quot;number of species&quot;</span><span class="p">:</span><span class="n">species_corrected</span><span class="p">,</span>
    <span class="s2">&quot;number of genus&quot;</span><span class="p">:</span><span class="n">spd2</span><span class="o">.</span><span class="n">genus</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
    <span class="s2">&quot;number of families&quot;</span><span class="p">:</span><span class="n">spd2</span><span class="o">.</span><span class="n">fam</span><span class="o">.</span><span class="n">nunique</span><span class="p">(),</span>
<span class="p">}</span>

<span class="n">a_summary</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="o">.</span><span class="n">from_dict</span><span class="p">(</span><span class="n">summary</span><span class="p">,</span> <span class="n">orient</span><span class="o">=</span><span class="s2">&quot;index&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>


<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">7</span><span class="p">))</span>

<span class="n">gs</span> <span class="o">=</span> <span class="n">GridSpec</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>

<span class="n">axone</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">])</span>
<span class="n">axtwo</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">:])</span>
<span class="n">axthree</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">6</span><span class="p">:,</span> <span class="mi">6</span><span class="p">:])</span>

<span class="c1"># axone</span>
<span class="n">table_one</span> <span class="o">=</span> <span class="n">bot</span><span class="o">.</span><span class="n">a_simple_formatted_table</span><span class="p">(</span><span class="n">axone</span><span class="p">,</span><span class="n">data</span><span class="p">,</span><span class="n">colLabels</span><span class="o">=</span><span class="n">colLabels</span><span class="p">,</span> <span class="n">colWidths</span><span class="o">=</span><span class="n">colWidths</span><span class="p">,</span> <span class="n">bbox</span><span class="o">=</span><span class="n">bbox</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>

<span class="c1"># axtwo</span>
<span class="n">locator</span> <span class="o">=</span> <span class="n">mdates</span><span class="o">.</span><span class="n">MonthLocator</span><span class="p">()</span>  <span class="c1"># every month</span>
<span class="n">fmt</span> <span class="o">=</span> <span class="n">mdates</span><span class="o">.</span><span class="n">DateFormatter</span><span class="p">(</span><span class="s1">&#39;%b&#39;</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">d1</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s2">&quot;date&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;species&quot;</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axtwo</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">locator</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_formatter</span><span class="p">(</span><span class="n">fmt</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;# species indentified&quot;</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="p">)</span>

<span class="c1"># axthree</span>
<span class="n">b</span> <span class="o">=</span> <span class="n">bot</span><span class="o">.</span><span class="n">a_single_column_table</span><span class="p">(</span><span class="n">axthree</span><span class="p">,</span> <span class="n">a_summary</span><span class="o">.</span><span class="n">values</span><span class="p">,</span> <span class="n">colWidths</span><span class="o">=</span><span class="p">[</span><span class="mf">.6</span><span class="p">,</span> <span class="mf">.4</span><span class="p">])</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_22_0.png" src="_images/analysis_22_0.png" />
</div>
</div>
<p><em><strong>Figure 7 :</strong>  Summary statistics for the second year of surveys. Notice the survey period began in February rather than June and the significantly increase across all metrics. <strong>Left :</strong> The number of unique species, unique samples, and unique sites surveyed per month in year 1. <strong>Top right</strong>: The number of species identified per survey per month in year 1. <strong>Bottom right</strong>: Summary statistics of year 1.</em></p>
<div class="section" id="categorizing-the-species">
<h3><span class="section-number">1.5.1. </span>Categorizing the species üåº<a class="headerlink" href="#categorizing-the-species" title="Permalink to this headline">¬∂</a></h3>
<p>The plants in this anaysis can be divided into native and non-native plants; all species that arrived after 1500 (the discovery of America) are considered neophytes. All other species are considered native and are evaluated under the IUCN Redlist. Nearly all native species have their conservation status characterized by an IUCN Redlist attribute and neophytes can be characterized by their potential (or lack thereof) to become invasive (there are culturally valuable neophytes but these account for 5 observations, or .08% of all species identified which fall under neither category). Some 70% of species identified were native and classified as ‚ÄúLeast Concern‚Äù, a proportion that remained stable throughout the survey period. This is in line with what might be expected: the survey would likely identify populations of hardy, common plants resistant to anthropogenic pressures with a significant minority of neophytes that are planted purposefully, spill over from gardens, or have become naturalized in the man-made landscape. Note that for all taxonomy and classification this analysis follows the designations of the national authority of Switzerland, Infoflora.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="k">def</span> <span class="nf">periodic_value_counts</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="s2">&quot;month&quot;</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="p">[]):</span>
    <span class="n">a</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="o">~</span><span class="n">data</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">exclude</span><span class="p">)]</span>
    <span class="n">b</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">a</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">period</span><span class="p">)[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="n">column</span><span class="p">:</span><span class="s2">&quot;data&quot;</span><span class="p">})</span>
    
    <span class="k">return</span> <span class="n">b</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>

<span class="k">def</span> <span class="nf">assign_axis_values</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="s2">&quot;month&quot;</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="s2">&quot;allstats&quot;</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="s2">&quot;NT&quot;</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="p">[],</span> <span class="n">bars</span><span class="o">=</span><span class="p">{}):</span>
    
    <span class="n">a_dates</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">==</span> <span class="n">value</span><span class="p">][</span><span class="n">period</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span>
    <span class="n">add_dates</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">these_dates</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">a_dates</span><span class="p">]</span>
    <span class="n">add_these</span> <span class="o">=</span> <span class="p">[{</span><span class="n">period</span><span class="p">:</span><span class="n">x</span><span class="p">,</span> <span class="n">column</span><span class="p">:</span><span class="n">aval</span><span class="p">,</span> <span class="s2">&quot;data&quot;</span><span class="p">:</span><span class="mi">0</span><span class="p">}</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">add_dates</span><span class="p">]</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">add_these</span><span class="p">)</span>
    <span class="n">bars</span><span class="o">.</span><span class="n">update</span><span class="p">({</span><span class="n">aval</span><span class="p">:{</span><span class="s2">&quot;data&quot;</span><span class="p">:</span><span class="n">data</span><span class="p">[</span><span class="n">data</span><span class="p">[</span><span class="n">column</span><span class="p">]</span> <span class="o">==</span> <span class="n">aval</span><span class="p">]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;month&quot;</span><span class="p">)[</span><span class="s2">&quot;data&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">}})</span>
    
    <span class="k">return</span> <span class="n">data</span><span class="p">,</span> <span class="n">bars</span>

<span class="c1"># count the number of occurences for each category at each survey</span>
<span class="n">spdas1</span> <span class="o">=</span> <span class="n">spd_allstats</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">spd_allstats</span><span class="p">[</span><span class="s2">&quot;year&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">spdas2</span> <span class="o">=</span> <span class="n">spd_allstats</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">spd_allstats</span><span class="p">[</span><span class="s2">&quot;year&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">these_years</span> <span class="o">=</span> <span class="p">[</span><span class="n">spdas1</span><span class="p">,</span> <span class="n">spdas2</span><span class="p">]</span>
<span class="n">these_dates</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="p">[</span><span class="s2">&quot;month&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">sort_values</span><span class="p">()</span><span class="o">.</span><span class="n">unique</span><span class="p">()</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">these_years</span><span class="p">]</span>
<span class="n">set_dates</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">these_dates</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span> <span class="o">|</span> <span class="nb">set</span><span class="p">(</span><span class="n">these_dates</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>

<span class="n">these_lists</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;allstats&quot;</span><span class="p">]</span>
<span class="n">not_these</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;X&quot;</span><span class="p">,</span><span class="s2">&quot;-&quot;</span><span class="p">]</span>

<span class="n">period</span> <span class="o">=</span> <span class="s2">&quot;month&quot;</span>
<span class="n">column</span> <span class="o">=</span> <span class="s2">&quot;allstats&quot;</span>

<span class="n">biogeo</span> <span class="o">=</span> <span class="s2">&quot;mp&quot;</span>

<span class="n">a_palette</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;LC&quot;</span><span class="p">:</span><span class="s2">&quot;dodgerblue&quot;</span><span class="p">,</span>
    <span class="s2">&quot;EN&quot;</span><span class="p">:</span><span class="s2">&quot;crimson&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NT&quot;</span><span class="p">:</span><span class="s2">&quot;magenta&quot;</span><span class="p">,</span>
    <span class="s2">&quot;VU&quot;</span><span class="p">:</span> <span class="s2">&quot;plum&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NE&quot;</span><span class="p">:</span><span class="s2">&quot;darkviolet&quot;</span><span class="p">,</span>
    <span class="s2">&quot;DD&quot;</span><span class="p">:</span><span class="s2">&quot;grey&quot;</span><span class="p">,</span>
    <span class="s2">&quot;WL&quot;</span><span class="p">:</span><span class="s2">&quot;red&quot;</span><span class="p">,</span>
    <span class="s2">&quot;BL&quot;</span><span class="p">:</span><span class="s2">&quot;black&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ongen&quot;</span><span class="p">:</span><span class="s2">&quot;fuchsia&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ns&quot;</span><span class="p">:</span><span class="s2">&quot;violet&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ni&quot;</span><span class="p">:</span><span class="s2">&quot;indigo&quot;</span>
<span class="p">}</span>
    

<span class="c1"># make bar for bar plots</span>
<span class="c1"># the unique values in the column of interest</span>
<span class="n">e</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">|</span> <span class="nb">set</span><span class="p">(</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">e</span><span class="o">=</span><span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">e</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">not_these</span><span class="p">]</span>

<span class="c1"># year one</span>
<span class="n">newDf</span> <span class="o">=</span> <span class="n">spdas1</span><span class="p">[(</span><span class="n">spdas1</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_as_yone</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">red_list_yone_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_yone</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_as_yone</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">red_list_yone_bars</span><span class="p">)</span>

<span class="c1"># year two</span>
<span class="n">newDf2</span> <span class="o">=</span> <span class="n">spdas2</span><span class="p">[(</span><span class="n">spdas2</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_as_ytwo</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf2</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">red_list_ytwo_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_ytwo</span><span class="p">,</span><span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_as_ytwo</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">red_list_ytwo_bars</span><span class="p">)</span>
    

<span class="c1"># rmp status per survey</span>
<span class="n">all_stat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">spd_allstats</span><span class="p">[</span><span class="o">~</span><span class="n">spd_allstats</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">)]</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;loc_date&quot;</span><span class="p">,</span><span class="s2">&quot;date&quot;</span><span class="p">])</span><span class="o">.</span><span class="n">allstats</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="n">column</span><span class="p">:</span><span class="s2">&quot;data&quot;</span><span class="p">})</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>

<span class="n">tick_spacing</span> <span class="o">=</span> <span class="mi">1</span>

<span class="n">full_name</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;LC&quot;</span><span class="p">:</span><span class="s2">&quot;Least Concern&quot;</span><span class="p">,</span>
    <span class="s2">&quot;EN&quot;</span><span class="p">:</span><span class="s2">&quot;Endangered&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NT&quot;</span><span class="p">:</span><span class="s2">&quot;Near threatened&quot;</span><span class="p">,</span>
    <span class="s2">&quot;VU&quot;</span><span class="p">:</span> <span class="s2">&quot;Vunerable&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NE&quot;</span><span class="p">:</span><span class="s2">&quot;Near endangered&quot;</span><span class="p">,</span>
    <span class="s2">&quot;DD&quot;</span><span class="p">:</span><span class="s2">&quot;Data deficient&quot;</span><span class="p">,</span>
    <span class="s2">&quot;WL&quot;</span><span class="p">:</span><span class="s2">&quot;Watchlist&quot;</span><span class="p">,</span>
    <span class="s2">&quot;BL&quot;</span><span class="p">:</span><span class="s2">&quot;Blacklist&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ongen&quot;</span><span class="p">:</span><span class="s2">&quot;Only genus ID&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ns&quot;</span><span class="p">:</span><span class="s2">&quot;Culturally Significant&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ni&quot;</span><span class="p">:</span><span class="s2">&quot;Neophyte&quot;</span>
<span class="p">}</span>

<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">14</span><span class="p">,</span><span class="mi">12</span><span class="p">))</span>

<span class="n">gs</span> <span class="o">=</span> <span class="n">GridSpec</span><span class="p">(</span><span class="mi">14</span><span class="p">,</span><span class="mi">12</span><span class="p">)</span>

<span class="n">axone</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">6</span><span class="p">])</span>
<span class="n">axtwo</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span> <span class="mi">7</span><span class="p">:])</span>
<span class="n">axthree</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">7</span><span class="p">:,</span> <span class="p">:])</span>

<span class="n">bot</span><span class="o">.</span><span class="n">a_stacked_bar_chart</span><span class="p">(</span><span class="n">axone</span><span class="p">,</span> <span class="n">red_list_yone_bars</span><span class="p">,</span> <span class="n">xaxis</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">)</span>
<span class="n">bot</span><span class="o">.</span><span class="n">a_stacked_bar_chart</span><span class="p">(</span><span class="n">axtwo</span><span class="p">,</span> <span class="n">red_list_ytwo_bars</span><span class="p">,</span> <span class="n">xaxis</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">)</span>

<span class="n">xlabel</span> <span class="o">=</span> <span class="s2">&quot;Month&quot;</span>
<span class="n">ylabel</span> <span class="o">=</span> <span class="s2">&quot;Number of species&quot;</span>

<span class="n">axone</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">ticker</span><span class="o">.</span><span class="n">MultipleLocator</span><span class="p">(</span><span class="n">tick_spacing</span><span class="p">))</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">ylabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="n">xlabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;2020&quot;</span><span class="p">)</span>
<span class="n">h</span><span class="p">,</span><span class="n">l</span> <span class="o">=</span> <span class="n">axone</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>
<span class="n">nl</span> <span class="o">=</span> <span class="p">[</span><span class="n">full_name</span><span class="p">[</span><span class="n">x</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">l</span><span class="p">]</span>
<span class="n">axone</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">h</span><span class="p">,</span><span class="n">nl</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">ticker</span><span class="o">.</span><span class="n">MultipleLocator</span><span class="p">(</span><span class="n">tick_spacing</span><span class="p">))</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="n">xlabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;2021&quot;</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">all_stat</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s2">&quot;date&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;data&quot;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;allstats&quot;</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axthree</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">ylabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="p">)</span>

<span class="n">h</span><span class="p">,</span><span class="n">l</span> <span class="o">=</span> <span class="n">axthree</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>
<span class="n">nl</span> <span class="o">=</span> <span class="p">[</span><span class="n">full_name</span><span class="p">[</span><span class="n">x</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">l</span><span class="p">]</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">h</span><span class="p">,</span><span class="n">nl</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>   
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
</pre></div>
</div>
<img alt="_images/analysis_25_1.png" src="_images/analysis_25_1.png" />
</div>
</div>
<p><em><strong>Figure 8 :</strong> Categorization of the identified species, plotted chronologically. <strong>Top left :</strong> The number of plants identified per month for 2020 surveys, <strong>Top right :</strong> The number of plants identified per month for the 2021 surveys. <strong>Bottom :</strong> The number of unique species identified per survey plotted monthly.</em></p>
</div>
<div class="section" id="non-invasive-neophytes">
<h3><span class="section-number">1.5.2. </span>Non-invasive neophytes üå∑<a class="headerlink" href="#non-invasive-neophytes" title="Permalink to this headline">¬∂</a></h3>
<p>The vast majority of neophytes have little chance of establishing spontaneous populations in the wild or even in urban spaces, yet a number have succesfully established populations in Switzerland over the past 500 years. One common non-invasive neophyte (although it can become invasive locally) is the annual <em>Conyza canadensis</em> (<em>Erigeron canadensis</em>) from North America that thrives in disturbed areas around the city. It was identified in 12 locations in August and September 2021, however these numbers underplay the relative amount of the plant in a given location as it spreads quickly and can dominate a given area during the late summer. Despite this dominance, this pioneer plant generally cedes its place to normal succession processes and thus is not considered invasive. Another two common examples of neophytes that have succesfully naturalized without becoming invasive are <em>Onobrychis viciifolia</em>, common in fields and along road verges in late spring and early summer, and <em>Eupatorium cannabinum</em>, common along road verges and river banks.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">images_to_display</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;resources/images/intro-1_scaled.jpg&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/20210822_192453_scaled.jpg&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/Selection_028.png&#39;</span>
    
<span class="p">]</span>

<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s2">&quot;Onobrychis viciifolia&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Conyza canadensis&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Eupatorium cannabinum&quot;</span><span class="p">,</span>
    
<span class="p">]</span>

<span class="n">rows</span> <span class="o">=</span> <span class="mi">1</span>
<span class="n">cols</span> <span class="o">=</span> <span class="mi">3</span>
<span class="n">fig_size</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>

<span class="c1"># open the images</span>
<span class="n">anarray</span><span class="o">=</span><span class="n">open_these_images</span><span class="p">(</span><span class="n">images_to_display</span><span class="p">)</span>

<span class="c1"># display the images</span>
<span class="n">display_grid_of_images</span><span class="p">(</span><span class="n">anarray</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="n">cols</span><span class="p">,</span> <span class="n">rows</span><span class="o">=</span><span class="n">rows</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="n">labels</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="n">fig_size</span> <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_28_0.png" src="_images/analysis_28_0.png" />
</div>
</div>
<p><em><strong>Figure 9 :</strong> Three common non invasive neophytes that grow in different environments around Biel/Bienne. <strong>Left to right :</strong> O. viciifolia is commonly found in fields, road and railroad berms in late spring and is often purposefully planted; C. canadensis is an annual common in lots and highly disturbed areas with fully spontaneous populations; E. cannabinum thrives along river banks and in more humid areas and can bloom throught the late spring and summer.</em></p>
</div>
<div class="section" id="invasive-species">
<h3><span class="section-number">1.5.3. </span>Invasive species  üå∫<a class="headerlink" href="#invasive-species" title="Permalink to this headline">¬∂</a></h3>
<p>A small number of neophytes are considered invasive, meaning they reproduce in the wild and pose a threat to humans and/or local ecosystems. Usually for invasive plants this is through displacement of plant populations - the invasive species takes over a given area and native species have difficulty pushing it out. Info Flora maintains a list of the current watch list and black list species, a number of which were identified over the course of the survey. Some of them were purposefully planted, such as the rows of <em>Mahonia aquifolium</em> throughout Biel/Bienne‚Äôs Madretsch quarter. Others have moved in of their own accord such as <em>Solidago canadensis</em> and <em>Erigeron Annuus</em>.</p>
<p>Invasive species in this study are defined as those species classified as either being on the <em>Watch List</em> (WL) or the <em>Black List</em> (BL) from the 2014 list compiled by Infoflora and the FOEN. Many more instances of BL species were recorded than WL species throughout the course of the survey period, as can be seen in the below chart and tables. In both cases, there are a few key invasive species that drive the dynamic, namely <em>Erigeron annuus</em> (BL), <em>Solidago canadensis</em> (BL), <em>Senecio inaequidens</em> (BL), <em>Buddleja davidii</em> (BL) and <em>Mahonia Aquifolium</em> (WL), with <em>E. annuus</em> twice as common as its nearest ‚Äúcompetitor‚Äù, <em>S. canadensis.</em></p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">images_to_display</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;resources/images/DSC02758_scaled.JPG&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/iledelasuze2_scaled.JPG&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/20210404_093051_scaled.jpg&#39;</span>
    
<span class="p">]</span>

<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s2">&quot;Madretschkanal - Solidago sp.&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Sch√ºssinsel - Erigeron annuus&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Kreuzplatz - Mahonia Aquifolium&quot;</span><span class="p">,</span>
    
<span class="p">]</span>

<span class="n">rows</span> <span class="o">=</span> <span class="mi">1</span>
<span class="n">cols</span> <span class="o">=</span> <span class="mi">3</span>
<span class="n">fig_size</span><span class="o">=</span><span class="p">(</span><span class="mi">16</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>

<span class="c1"># open the images</span>
<span class="n">anarray</span><span class="o">=</span><span class="n">open_these_images</span><span class="p">(</span><span class="n">images_to_display</span><span class="p">)</span>

<span class="c1"># display the images</span>
<span class="n">display_grid_of_images</span><span class="p">(</span><span class="n">anarray</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="n">cols</span><span class="p">,</span> <span class="n">rows</span><span class="o">=</span><span class="n">rows</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="n">labels</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="n">fig_size</span> <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_31_0.png" src="_images/analysis_31_0.png" />
</div>
</div>
<p><em><strong>Figure 10 :</strong> Three common species of invasive plants found throughout the city. <strong>From left to right:</strong> S. canadensis, common throughout the city in both disturbed areas and gardens (homeowners seem to like the yellow flowers); E. annuus which can be found flowering throughout the city from June to September;  M. aquifolium, unlike most cases of the first two species, was planted purposefully throughout the city and has spread spontaneously through consumption of its berries by birds.</em></p>
<p>One interesting absence across all survey locations was <em>Reynoutra japonica</em>, a member of the Blacklist (BL). An aggressive invader of riverbanks and increasingly common throughout waterways in Switzerland, only one stand was identified near the Mettmoos park in L√§ngholz (close to an official survey area). Additionally, the species <em>Prunus Laurocerasus</em> was undercounted as it is primaly a plant of hedgerows, which were generally not surveyed. As such, a significantly higher amount of prunus laurocerasus is encountered in Biel.</p>
<p>Note as well that ‚Äúinvasive‚Äù describes a behavior, namely the potential to dominate certain areas and preventing other plants from growing. In highly disturbed areas or in ecological systems in disorder it is possible for native species to act invasively and for typically non-invasive neophytes to become locally invasive. Thus, the species on the invasive species list in some sense undercount the invasive status as it does not take locally invasive species into account.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Format these two tables</span>
<span class="n">a</span> <span class="o">=</span> <span class="n">spd</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">spd</span><span class="o">.</span><span class="n">inv</span> <span class="o">==</span> <span class="s2">&quot;BL&quot;</span><span class="p">,</span> <span class="s2">&quot;species&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
<span class="n">b</span> <span class="o">=</span> <span class="n">spd</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">spd</span><span class="o">.</span><span class="n">inv</span> <span class="o">==</span> <span class="s2">&quot;WL&quot;</span><span class="p">,</span> <span class="s2">&quot;species&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>

<span class="c1"># tables</span>
<span class="n">black_list</span> <span class="o">=</span> <span class="p">[[</span><span class="n">a</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">x</span><span class="p">],</span><span class="n">a</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="n">x</span><span class="p">]]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">a</span><span class="p">))]</span>
<span class="n">white_list</span> <span class="o">=</span> <span class="p">[[</span><span class="n">b</span><span class="o">.</span><span class="n">index</span><span class="p">[</span><span class="n">x</span><span class="p">],</span> <span class="n">b</span><span class="o">.</span><span class="n">values</span><span class="p">[</span><span class="n">x</span><span class="p">]]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">np</span><span class="o">.</span><span class="n">arange</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">b</span><span class="p">))]</span>

<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mf">8.5</span><span class="p">,</span><span class="mi">9</span><span class="p">))</span>

<span class="n">gs</span> <span class="o">=</span> <span class="n">GridSpec</span><span class="p">(</span><span class="mi">14</span><span class="p">,</span><span class="mi">9</span><span class="p">)</span>

<span class="n">axone</span><span class="o">=</span><span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:,</span> <span class="p">:</span><span class="mi">4</span><span class="p">])</span>
<span class="n">axtwo</span><span class="o">=</span><span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span><span class="mi">5</span><span class="p">:])</span>
<span class="n">atable</span> <span class="o">=</span> <span class="n">bot</span><span class="o">.</span><span class="n">a_simple_formatted_table</span><span class="p">(</span><span class="n">axone</span><span class="p">,</span> <span class="n">black_list</span><span class="p">,</span> <span class="n">colLabels</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;species&quot;</span><span class="p">,</span> <span class="s2">&quot;sightings&quot;</span><span class="p">],</span> <span class="n">colWidths</span><span class="o">=</span><span class="p">[</span><span class="mf">.7</span><span class="p">,</span><span class="mf">.3</span><span class="p">],</span> <span class="n">bbox</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">])</span>
<span class="n">axone</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;BlackList&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">16</span><span class="p">)</span>

<span class="n">atable2</span> <span class="o">=</span> <span class="n">bot</span><span class="o">.</span><span class="n">a_simple_formatted_table</span><span class="p">(</span><span class="n">axtwo</span><span class="p">,</span> <span class="n">white_list</span><span class="p">,</span> <span class="n">colLabels</span><span class="o">=</span><span class="p">[</span><span class="s2">&quot;species&quot;</span><span class="p">,</span> <span class="s2">&quot;sightings&quot;</span><span class="p">],</span> <span class="n">colWidths</span><span class="o">=</span><span class="p">[</span><span class="mf">.7</span><span class="p">,</span><span class="mf">.3</span><span class="p">],</span> <span class="n">bbox</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">])</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">axis</span><span class="p">(</span><span class="s2">&quot;off&quot;</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;WatchList&quot;</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">16</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_34_0.png" src="_images/analysis_34_0.png" />
</div>
</div>
<p><em><strong>Figure 11:</strong> The above tables of the Black List and Watch List species demonstrate that the overall abundance of invasive species in the survey locations was driven by a few key species. Part of this is an artefact of sampling certain areas and not sampling the hedgerows in which Prunus laurocerasus is highly present.</em></p>
<p>In most cases either no invasives or only one invasive species was found in each survey. This is constant across months and across years. Since only a handful of invasives are driving these numbres, it is clear to that the higher numbers of invasive species identified in June, July, August and September corresponds to the flowering season of <em>E. Annuus</em>, <em>S. canadensis</em> and <em>S. inaequidens</em></p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">these_lists</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;allstats&quot;</span><span class="p">]</span>

<span class="n">period</span> <span class="o">=</span> <span class="s2">&quot;month&quot;</span>
<span class="n">column</span> <span class="o">=</span> <span class="s2">&quot;allstats&quot;</span>

<span class="n">biogeo</span> <span class="o">=</span> <span class="s2">&quot;mp&quot;</span>

<span class="n">a_palette</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;LC&quot;</span><span class="p">:</span><span class="s2">&quot;dodgerblue&quot;</span><span class="p">,</span>
    <span class="s2">&quot;EN&quot;</span><span class="p">:</span><span class="s2">&quot;crimson&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NT&quot;</span><span class="p">:</span><span class="s2">&quot;magenta&quot;</span><span class="p">,</span>
    <span class="s2">&quot;VU&quot;</span><span class="p">:</span> <span class="s2">&quot;plum&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NE&quot;</span><span class="p">:</span><span class="s2">&quot;darkviolet&quot;</span><span class="p">,</span>
    <span class="s2">&quot;DD&quot;</span><span class="p">:</span><span class="s2">&quot;grey&quot;</span><span class="p">,</span>
    <span class="s2">&quot;WL&quot;</span><span class="p">:</span><span class="s2">&quot;red&quot;</span><span class="p">,</span>
    <span class="s2">&quot;BL&quot;</span><span class="p">:</span><span class="s2">&quot;black&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ongen&quot;</span><span class="p">:</span><span class="s2">&quot;fuchsia&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ns&quot;</span><span class="p">:</span><span class="s2">&quot;violet&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ni&quot;</span><span class="p">:</span><span class="s2">&quot;indigo&quot;</span>
<span class="p">}</span>    

<span class="c1"># make bar for bar plots</span>
<span class="c1"># the unique values in the column of interest</span>
<span class="n">e</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">|</span> <span class="nb">set</span><span class="p">(</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">e</span><span class="o">=</span><span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">e</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">not_these</span><span class="p">]</span>

<span class="c1"># year one</span>
<span class="n">newDf</span> <span class="o">=</span> <span class="n">spdas1</span><span class="p">[(</span><span class="n">spdas1</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_as_yone</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">red_list_yone_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_yone</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_as_yone</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">red_list_yone_bars</span><span class="p">)</span>

<span class="c1"># year two</span>
<span class="n">newDf2</span> <span class="o">=</span> <span class="n">spdas2</span><span class="p">[(</span><span class="n">spdas2</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_as_ytwo</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf2</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">red_list_ytwo_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_ytwo</span><span class="p">,</span><span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_as_ytwo</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">red_list_ytwo_bars</span><span class="p">)</span>

<span class="c1"># rmp status per survey</span>
<span class="n">all_stat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">spd_allstats</span><span class="p">[</span><span class="o">~</span><span class="n">spd_allstats</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">)]</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;loc_date&quot;</span><span class="p">,</span><span class="s2">&quot;date&quot;</span><span class="p">])</span><span class="o">.</span><span class="n">allstats</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="n">column</span><span class="p">:</span><span class="s2">&quot;data&quot;</span><span class="p">})</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
</pre></div>
</div>
</div>
</div>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">these_lists</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;inv&quot;</span><span class="p">]</span>

<span class="n">period</span> <span class="o">=</span> <span class="s2">&quot;month&quot;</span>
<span class="n">column</span> <span class="o">=</span> <span class="s2">&quot;inv&quot;</span>

<span class="n">biogeo</span> <span class="o">=</span> <span class="s2">&quot;mp&quot;</span>

<span class="n">a_palette</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;WL&quot;</span><span class="p">:</span><span class="s2">&quot;dodgerblue&quot;</span><span class="p">,</span>
    <span class="s2">&quot;BL&quot;</span><span class="p">:</span><span class="s2">&quot;crimson&quot;</span><span class="p">,</span>
<span class="p">}</span>    

<span class="c1"># make bar for bar plots</span>
<span class="c1"># the unique values in the column of interest</span>
<span class="n">e</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">|</span> <span class="nb">set</span><span class="p">(</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">e</span><span class="o">=</span><span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">e</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">not_these</span><span class="p">]</span>

<span class="c1"># year one</span>
<span class="n">newDf</span> <span class="o">=</span> <span class="n">spdas1</span><span class="p">[(</span><span class="n">spdas1</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_rch_yone</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">inv_yone_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_yone</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_rch_yone</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">inv_yone_bars</span><span class="p">)</span>

<span class="c1"># year two</span>
<span class="n">newDf2</span> <span class="o">=</span> <span class="n">spdas2</span><span class="p">[(</span><span class="n">spdas2</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_rch_ytwo</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf2</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">inv_ytwo_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_ytwo</span><span class="p">,</span><span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_rch_ytwo</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">inv_ytwo_bars</span><span class="p">)</span>

<span class="c1"># rmp status per survey</span>
<span class="n">inv_stat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">spd</span><span class="p">[</span><span class="o">~</span><span class="n">spd</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">)]</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;loc_date&quot;</span><span class="p">,</span><span class="s2">&quot;date&quot;</span><span class="p">])</span><span class="o">.</span><span class="n">inv</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="n">column</span><span class="p">:</span><span class="s2">&quot;data&quot;</span><span class="p">})</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>

<span class="n">tick_spacing</span> <span class="o">=</span> <span class="mi">1</span>

<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">8</span><span class="p">))</span>

<span class="n">gs</span> <span class="o">=</span> <span class="n">GridSpec</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>

<span class="n">axone</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">])</span>
<span class="n">axtwo</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">:])</span>
<span class="n">axthree</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">6</span><span class="p">:,</span> <span class="p">:])</span>

<span class="n">bot</span><span class="o">.</span><span class="n">a_stacked_bar_chart</span><span class="p">(</span><span class="n">axone</span><span class="p">,</span> <span class="n">inv_yone_bars</span><span class="p">,</span> <span class="n">xaxis</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">)</span>
<span class="n">bot</span><span class="o">.</span><span class="n">a_stacked_bar_chart</span><span class="p">(</span><span class="n">axtwo</span><span class="p">,</span> <span class="n">inv_ytwo_bars</span><span class="p">,</span> <span class="n">xaxis</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">)</span>

<span class="n">xlabel</span> <span class="o">=</span> <span class="s2">&quot;Month&quot;</span>
<span class="n">ylabel</span> <span class="o">=</span> <span class="s2">&quot;Number of species&quot;</span>

<span class="n">axone</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">ticker</span><span class="o">.</span><span class="n">MultipleLocator</span><span class="p">(</span><span class="n">tick_spacing</span><span class="p">))</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">ylabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="n">xlabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;2020&quot;</span><span class="p">)</span>

<span class="n">h</span><span class="p">,</span><span class="n">l</span> <span class="o">=</span> <span class="n">axone</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>

<span class="n">axone</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">h</span><span class="p">,</span> <span class="p">[</span><span class="s2">&quot;Watchlist&quot;</span><span class="p">,</span> <span class="s2">&quot;Blacklist&quot;</span><span class="p">])</span>

<span class="n">axtwo</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">ticker</span><span class="o">.</span><span class="n">MultipleLocator</span><span class="p">(</span><span class="n">tick_spacing</span><span class="p">))</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="n">xlabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;2021&quot;</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">inv_stat</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s2">&quot;date&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;data&quot;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;inv&quot;</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axthree</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">ylabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">h</span><span class="p">,</span> <span class="p">[</span><span class="s2">&quot;Watchlist&quot;</span><span class="p">,</span> <span class="s2">&quot;Blacklist&quot;</span><span class="p">])</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>   
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
</pre></div>
</div>
<img alt="_images/analysis_38_1.png" src="_images/analysis_38_1.png" />
</div>
</div>
<p><strong>Figure 11 :</strong> The number of invasives identified over the course of the survey. <strong>Top :</strong> The number of invasives identified per month in 2020 and 2021: <strong>Bottom :</strong> The number of invasive species identified per survey over the course of two years.</p>
</div>
<div class="section" id="redlist-and-priority-list-species-on-the-middle-plateau">
<h3><span class="section-number">1.5.4. </span>Redlist and priority list species on the middle plateau  üå∏<a class="headerlink" href="#redlist-and-priority-list-species-on-the-middle-plateau" title="Permalink to this headline">¬∂</a></h3>
<p>The IUCN Redlist is an internationally comparable ranking of the conservation status of individual species and biotopes. Infoflora worked with the IUCN to create a national redlist for Switzerland, last updated in 2016. This analysis was complemented in 2019 with the publishing of a regional Redlist report. This provided a IUCN redlist rating for each native species in each of Switzerland‚Äôs main biogeographical regions. In the previous analysis the redlist categories were based on the national rating as about 600 observations (out of 7600) occured outside of the Central Plateau region. This section uses only observations made within the middle plateau region and categorizes them according to this regional redlist analysis. This is still the vast majority of survey locations and species identified and as will be seen does not change the overall decomposition. As noted in the introduction, the regional redlist for the middle plateau places many species on a higher risk rating than the national redlist. While species in each category were identified, the vast majority of native species were of least concern. Compared to using the national redlist ranking, the overall results do not change, with only a few species moving from Least Concern to Near Threatened.</p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># Images for invasives</span>

<span class="n">images_to_display</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;resources/images/dipsacus-pilosela_scaled.JPG&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/20210404_154240_scaled.jpg&#39;</span><span class="p">,</span>
    <span class="s1">&#39;resources/images/epilobium-dodonaei_scaled.JPG&#39;</span>    
<span class="p">]</span>

<span class="n">labels</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s2">&quot;Sch√ºessinsel - Dispacus pilosus&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Portplatz - Anemone nemorosa&quot;</span><span class="p">,</span>
    <span class="s2">&quot;Suze, B√ºrenstrasse - Epilobium dodonaei&quot;</span>    
<span class="p">]</span>

<span class="n">cols</span><span class="o">=</span> <span class="mi">1</span>
<span class="n">rows</span><span class="o">=</span> <span class="mi">3</span>
<span class="n">fig_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">16</span><span class="p">,</span><span class="mi">8</span><span class="p">)</span>



<span class="n">anarray</span> <span class="o">=</span> <span class="n">open_these_images</span><span class="p">(</span><span class="n">images_to_display</span><span class="p">)</span>

<span class="n">display_grid_of_images</span><span class="p">(</span><span class="n">anarray</span><span class="p">,</span> <span class="n">cols</span><span class="o">=</span><span class="n">cols</span><span class="p">,</span> <span class="n">rows</span><span class="o">=</span><span class="n">rows</span><span class="p">,</span> <span class="n">labels</span><span class="o">=</span><span class="n">labels</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="n">fig_size</span> <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="_images/analysis_41_0.png" src="_images/analysis_41_0.png" />
</div>
</div>
<p><em><strong>Figure 12 :</strong> Some examples of native species found throughout Biel/Bienne. <strong>From top to bottom:</strong> Dipsacus Pilosus, a regionally near threatened species growing along the Suze; Anemone nemorosa a common woodland species of least concern that grows under some city hedges; Epilobium dodonaei another regionally near threatened species growing along a different section of the Suze river</em></p>
<div class="cell tag_hide-input docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">these_lists</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;rch&quot;</span><span class="p">,</span><span class="s2">&quot;ni&quot;</span><span class="p">,</span> <span class="s2">&quot;inv&quot;</span><span class="p">,</span> <span class="s2">&quot;ongen&quot;</span><span class="p">]</span>


<span class="n">period</span> <span class="o">=</span> <span class="s2">&quot;month&quot;</span>
<span class="n">column</span> <span class="o">=</span> <span class="s2">&quot;rmp&quot;</span>

<span class="n">biogeo</span> <span class="o">=</span> <span class="s2">&quot;mp&quot;</span>

<span class="n">a_palette</span> <span class="o">=</span> <span class="p">{</span>
    <span class="s2">&quot;LC&quot;</span><span class="p">:</span><span class="s2">&quot;dodgerblue&quot;</span><span class="p">,</span>
    <span class="s2">&quot;EN&quot;</span><span class="p">:</span><span class="s2">&quot;crimson&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NT&quot;</span><span class="p">:</span><span class="s2">&quot;magenta&quot;</span><span class="p">,</span>
    <span class="s2">&quot;VU&quot;</span><span class="p">:</span> <span class="s2">&quot;plum&quot;</span><span class="p">,</span>
    <span class="s2">&quot;NE&quot;</span><span class="p">:</span><span class="s2">&quot;darkviolet&quot;</span><span class="p">,</span>
    <span class="s2">&quot;DD&quot;</span><span class="p">:</span><span class="s2">&quot;grey&quot;</span><span class="p">,</span>
<span class="p">}</span>    

<span class="c1"># make bar for bar plots</span>
<span class="c1"># the unique values in the column of interest</span>
<span class="n">e</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="nb">set</span><span class="p">(</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">|</span> <span class="nb">set</span><span class="p">(</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">()))</span>
<span class="n">e</span><span class="o">=</span><span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">e</span> <span class="k">if</span> <span class="n">x</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">not_these</span><span class="p">]</span>

<span class="c1"># year one</span>
<span class="n">newDf</span> <span class="o">=</span> <span class="n">spdas1</span><span class="p">[(</span><span class="n">spdas1</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas1</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_rch_yone</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">red_list_yone_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_yone</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_rch_yone</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">red_list_yone_bars</span><span class="p">)</span>

<span class="c1"># year two</span>
<span class="n">newDf2</span> <span class="o">=</span> <span class="n">spdas2</span><span class="p">[(</span><span class="n">spdas2</span><span class="o">.</span><span class="n">biogeo</span> <span class="o">==</span> <span class="n">biogeo</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="o">~</span><span class="n">spdas2</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">))]</span>
<span class="n">data_rch_ytwo</span> <span class="o">=</span> <span class="n">periodic_value_counts</span><span class="p">(</span><span class="n">newDf2</span><span class="p">,</span> <span class="n">column</span><span class="p">,</span> <span class="n">exclude</span><span class="o">=</span><span class="n">not_these</span><span class="p">)</span>

<span class="n">red_list_ytwo_bars</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">aval</span> <span class="ow">in</span> <span class="n">e</span><span class="p">:</span>
    <span class="n">d_ytwo</span><span class="p">,</span><span class="n">_</span> <span class="o">=</span> <span class="n">assign_axis_values</span><span class="p">(</span><span class="n">data_rch_ytwo</span><span class="p">,</span> <span class="n">period</span><span class="o">=</span><span class="n">period</span><span class="p">,</span> <span class="n">column</span><span class="o">=</span><span class="n">column</span><span class="p">,</span> <span class="n">value</span><span class="o">=</span><span class="n">aval</span><span class="p">,</span> <span class="n">these_dates</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">bars</span><span class="o">=</span><span class="n">red_list_ytwo_bars</span><span class="p">)</span>
    
<span class="c1"># rmp status per survey</span>
<span class="n">rmp_stat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">spd</span><span class="p">[</span><span class="o">~</span><span class="n">spd</span><span class="p">[</span><span class="n">column</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">not_these</span><span class="p">)]</span><span class="o">.</span><span class="n">groupby</span><span class="p">([</span><span class="s2">&quot;loc_date&quot;</span><span class="p">,</span><span class="s2">&quot;date&quot;</span><span class="p">])</span><span class="o">.</span><span class="n">rmp</span><span class="o">.</span><span class="n">value_counts</span><span class="p">())</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="n">column</span><span class="p">:</span><span class="s2">&quot;data&quot;</span><span class="p">})</span><span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>

<span class="n">tick_spacing</span> <span class="o">=</span> <span class="mi">1</span>

<span class="n">fig</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">figure</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">8</span><span class="p">))</span>

<span class="n">gs</span> <span class="o">=</span> <span class="n">GridSpec</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">11</span><span class="p">)</span>

<span class="n">axone</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span> <span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">])</span>
<span class="n">axtwo</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">0</span><span class="p">:</span><span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">:])</span>
<span class="n">axthree</span> <span class="o">=</span> <span class="n">fig</span><span class="o">.</span><span class="n">add_subplot</span><span class="p">(</span><span class="n">gs</span><span class="p">[</span><span class="mi">6</span><span class="p">:,</span> <span class="p">:])</span>

<span class="n">bot</span><span class="o">.</span><span class="n">a_stacked_bar_chart</span><span class="p">(</span><span class="n">axone</span><span class="p">,</span> <span class="n">red_list_yone_bars</span><span class="p">,</span> <span class="n">xaxis</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">)</span>
<span class="n">bot</span><span class="o">.</span><span class="n">a_stacked_bar_chart</span><span class="p">(</span><span class="n">axtwo</span><span class="p">,</span> <span class="n">red_list_ytwo_bars</span><span class="p">,</span> <span class="n">xaxis</span><span class="o">=</span><span class="n">set_dates</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">)</span>

<span class="n">xlabel</span> <span class="o">=</span> <span class="s2">&quot;Month&quot;</span>
<span class="n">ylabel</span> <span class="o">=</span> <span class="s2">&quot;Number of species&quot;</span>

<span class="n">axone</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">ticker</span><span class="o">.</span><span class="n">MultipleLocator</span><span class="p">(</span><span class="n">tick_spacing</span><span class="p">))</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">ylabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="n">xlabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
<span class="n">axone</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;2020&quot;</span><span class="p">)</span>

<span class="n">h</span><span class="p">,</span><span class="n">l</span> <span class="o">=</span> <span class="n">axone</span><span class="o">.</span><span class="n">get_legend_handles_labels</span><span class="p">()</span>

<span class="n">ls</span> <span class="o">=</span> <span class="p">[</span><span class="n">full_name</span><span class="p">[</span><span class="n">x</span><span class="p">]</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">l</span><span class="p">]</span>
<span class="n">axone</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">h</span><span class="p">,</span><span class="n">ls</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">xaxis</span><span class="o">.</span><span class="n">set_major_locator</span><span class="p">(</span><span class="n">ticker</span><span class="o">.</span><span class="n">MultipleLocator</span><span class="p">(</span><span class="n">tick_spacing</span><span class="p">))</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="n">xlabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">12</span><span class="p">)</span>
<span class="n">axtwo</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="s2">&quot;2021&quot;</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">scatterplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">rmp_stat</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="s2">&quot;date&quot;</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="s2">&quot;data&quot;</span><span class="p">,</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;rmp&quot;</span><span class="p">,</span> <span class="n">palette</span><span class="o">=</span><span class="n">a_palette</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axthree</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">ylabel</span><span class="p">,</span> <span class="n">fontsize</span><span class="o">=</span><span class="mi">14</span><span class="p">,</span> <span class="n">labelpad</span><span class="o">=</span><span class="mi">14</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
<span class="n">axthree</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">h</span><span class="p">,</span><span class="n">ls</span><span class="p">,</span> <span class="n">loc</span><span class="o">=</span><span class="s2">&quot;center&quot;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>   
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
/tmp/ipykernel_18550/3455561462.py:12: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  data = data.append(add_these)
</pre></div>
</div>
<img alt="_images/analysis_43_1.png" src="_images/analysis_43_1.png" />
</div>
</div>
<p><em><strong>Figure 13 :</strong> The number of native species in the central plateau region identified over the course of the surveys classified by Redlist status. <strong>Top :</strong> The number of species in each Redlist category identified per month per year. <strong>Bottom :</strong>  The number of native species in each category identified per survey over the whole time period.</em></p>
<p>It must be taken into consideration that the genetic origin of many of these populations is unknown, especially for species that are widely commercially available. To use the species highlighted in Figure 12, both <em>A. nemorosa</em> and <em>E. dodonaie</em> are common garden plants and stand a good chance of being sourced from commercial seeds planted purposefully, while <em>D. Pilosus</em> is likely a remnant species since it is not easily available from a commercial source. This can only be resolved through genetic testing.</p>
</div>
</div>
<div class="section" id="discussion-and-conclusion">
<h2><span class="section-number">1.6. </span>Discussion and conclusion üíê<a class="headerlink" href="#discussion-and-conclusion" title="Permalink to this headline">¬∂</a></h2>
<p>These surveys add to the body of evidence that shows urban spaces can provide signficant plant diversity. Some general conclusions about the herbaceous flowering plants of Biel/Bienne can be drawn from the above data is the following:</p>
<ul class="simple">
<li><p>While invasive speces are widespread, they are characterized by a few key species</p></li>
<li><p>Neophytes form a critical part of the vegetation of the city</p></li>
<li><p>Only a few threatened or near threatened species are able to survive in the Biel/Bienne urban matrix.</p></li>
<li><p>Public spaces contain significantly more native vegetation (around 75%) than private gardens (30%)</p></li>
</ul>
<p>Plant surveys are best done year on year as weather and other factors can significantly affect the population distributions of vascular plant species. The first survey year, 2020, coincided with the warmest year on record in Switzerland, tied with 2018, with a scarcity of rainfall. Meanwhile, 2021 saw the wettest and coldest spring and summer weather for many local records. It is thus to be expected that these extreme variations would result in different plant species being identified across Biel/Bienne as a whole and even in certain survey locations. One anecdote of this is a regionally Near Threatened species, <em>Dipsacus pilosus</em>, which was identified near the Ile de la Suze on the banks of the Suze river in 2020, however in 2021 the site was submerged for much of the survey period due to the extra rainfall and there was no sign of its presence. While this sort of data can begin to be teased out by comparing the roughly 50 sites that overlap between 2020 and 21, the overall year on year analysis should wait until the 2022 data can be gathered.</p>
<p>This report does not delve into comparisons of features and does not make use of the fine scale of these surveys which would allow for a neighborhood by neighborhood analysis of plant resources in the city. Based on the data already keyed in to the dataset, there are useful analysis that can be done to compare different city features (such as railroad berms) and renaturalization sites (such as the Ile de la Suze) in and around Biel/Bienne. In addition, there rich sources of data available from municipal authorities that could allow explanatory causes to be determined and differentiated site treatments to be evaluated. Finally, digitizing the area component available in the survey notes will allow use of tools such as the Species Richness Model that ties plant diversity to greenspace size in urban areas.</p>
<p>To conclude, there are a number of interesting extensions to this analysis, some of which are possible with the available data and some of which will require integration with other datasets or collection of additional data. This report does not provide a causal analysis, test an existing thesis or build a predictive model, but of course these are the ultimate goals of the surveys. As such, notebooks related to these issues will be published in the coming months. Further, the plant surveys have begun anew and it is expected to match or exceed the observations made in 2021, which will provide a strong year on year data set of species make-up across seasons and across the city. There are a few modifications in the protocol which will facilite data collection and provide information on species density.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="intro.html" title="previous page">Biel‚Äôs flowers</a>
    <a class='right-next' id="next-link" href="germination.html" title="next page"><span class="section-number">2. </span>Wildflower germination</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By hammerdirt botanist<br/>
        
            &copy; Copyright 2021.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>